contents/tutorials/track-high-volume-apis.md

```diff
@@ -0,0 +1,266 @@
+    'sample',
+    req.params.id,
+    { 
+      sendFeatureFlagEvents: false,
+      personProperties: {
+        'firstRouteCalled': req.url,
+      },
+    }
+  );
+//...
+```
+
+In my small example, this cut response time by about 10ms, and every millisecond counts for high-volume APIs.
+
+## 2. Cache usage and batch events
+
+Another way to capture events for high-volume APIs is to cache usage of the API and then capture a batch of events at a time interval. 
+
+To do this, we use [node-cache](https://www.npmjs.com/package/node-cache) to create a basic cache to store the number of requests, then use [node-cron](https://www.npmjs.com/package/node-cron) to set up a cron job to read the cache, capture values for each of the IDs, and clear the cache. 
+
+First, install them both.
+
+```bash
+npm i node-cache node-cron
+```
+
+Next, set up our cache which simply gets the current value for the param ID and then increments it.
+
+```node
+// index.js
+//...
+const NodeCache = require('node-cache');
+const cache = new NodeCache()
+
+//...
+
+app.get('/big/:id', async (req, res) => {
+  let calls = cache.get(req.params.id) || 0;
+  calls++;
+  cache.set(req.params.id, calls);
+
+  res.send(`Welcome to the big route, ${req.params.id}`);
+})
+//...
+```
+
+After, we can set up our cron job to loop through the keys in the cache, get the number of events for that key, capture a batch event with a distinct ID (we use later), then clear the cache once completed.
+
+```node
+// index.js
+//...
+const cron = require('node-cron');
+
+//...
+
+cron.schedule('*/10 * * * * *', () => {
+  cache.keys().forEach((key) => {
+    let calls = cache.get(key);
+    client.capture(
+      { 
+        event: 'big route batch',
+        distinctId: key,
+        properties: { 
+          calls: calls,
+					distinctId: key
+        },
+      },
+    );
+  });
+  cache.flushAll();
+});
+```
+
+Once you run this and send multiple events to different IDs, you see batch events with calls as a property.
+
+![Batch](../images/tutorials/track-high-volume-apis/batch.png)
+
+## Insights with batched events
+
+Creating insights for batched or sampled events requires a little more planning than standard insights. This is because the actual values require one more step of setup. We rely on both formula mode and property aggregations to get the actual values we want for analysis.
+
+### Extrapolating sampled events with formula mode
+
+If we want sampled events to represent actual usage, we can multiply the rollout percentage to extrapolate a total count. For example, if we rolled out a flag to capture 10% of events, we can create an insight that takes the total count of events and multiplies it by 10.
+
+![Extrapolated](../images/tutorials/track-high-volume-apis/extrapolated.png)
+
+If you are doing this, it is important to make sure your rollout percentage matches your formula multiple to avoid large inaccuracies. Of course, any extrapolation isn’t perfectly accurate anyways.
```

Comment:
Not sure I would include the last sentence. It's slightly confusing and I dont think adding much value

```suggestion
If you are doing this, it is important to make sure your rollout percentage matches your formula multiple to avoid large inaccuracies.
```